pallets/gear/src/lib.rs

```diff
@@ -562,7 +562,29 @@ pub mod pallet {
                     )
                 };
 
-                core_processor::handle_journal(journal.clone(), &mut ext_manager);
+                // TODO: Check whether we charge gas fee for submitting code after #646
+                let notes = journal
+                    .into_iter()
+                    .filter_map(|note| -> Option<Result<JournalNote, _>> {
+                        match note {
+                            JournalNote::GasBurned { .. }
```

Comment:
Seems because of that note you will have to refactor `get_gas_spent` another time in #975, because later value of  `remaining_gas` variable will be None (in case we had a simple execution, when program doesn't send any message).